pkg/subctl/cmd/gather/logs.go

```diff
@@ -62,41 +61,62 @@ func gatherPodLogsByContainer(podLabelSelector, container string, info Info) {
 	}
 }
 
-func podLogsToFile(pod *corev1.Pod, podLogOptions *corev1.PodLogOptions, info Info) error {
+func getlogs(pod *corev1.Pod, podLogOptions *corev1.PodLogOptions, info Info) error {
+	podLogOptions.Previous = true
 	logRequest := info.ClientSet.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, podLogOptions)
+	logStream, _ := logRequest.Stream()  // TODO: Check for error other than "no previous pods found"
 
-	logs, err := processLogStream(logRequest)
+	// if no previous pods found, logstream == nil, ignore it
+	if logStream != nil {
+		info.Status.QueueWarningMessage(fmt.Sprintf("found previous instances of pod %s", pod.Name))
+		err := writeLogs(logStream, info, pod.Name, ".log.old")
+		if err != nil {
+			return err
+		}
+		defer logStream.Close()
+	}
+
+	// Running with Previous = false on the same pod
+	var err error
+	podLogOptions.Previous = false
+	logRequest = info.ClientSet.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, podLogOptions)
+	logStream, err = logRequest.Stream()
+	if err != nil {
+		return errors.WithMessage(err, "error opening log stream")
+	}
+	defer logStream.Close()
+	err = writeLogs(logStream, info, pod.Name, ".log")
+	if err != nil {}
+	return nil
```

Comment:
And this part another function:  "getExistingLogs()"

In such way we divide the function in smaller more meaningful chunks